src/views/virtualmachines/tree/utils/utils.tsx

```diff
@@ -26,11 +38,12 @@ export interface TreeViewDataItemWithHref extends TreeViewDataItem {
 
 const buildProjectMap = (
   getResourceUrl: GetResourceUrl,
-  vms: V1VirtualMachine[],
+  vms: MulticlusterResource<V1VirtualMachine>[] | V1VirtualMachine[],
   currentPageVMName: string,
   currentVMTab: string,
   treeViewDataMap: Record<string, TreeViewDataItemWithHref>,
   foldersEnabled: boolean,
+  cluster?: string,
```

Comment:
You can probably just use vm.cluster here, instead of passing in a cluster string.